mlos_core/mlos_core/tests/optimizers/optimizer_test.py

```diff
@@ -268,41 +266,39 @@ def objective(point: pd.DataFrame) -> pd.DataFrame:
             _LOG.debug("Optimizer is done with random init.")
 
         # loop for optimizer
-        suggestion = optimizer.suggest()
+        suggestion, metadata = optimizer.suggest()
         observation = objective(suggestion)
-        optimizer.register(suggestion, observation)
+        optimizer.register(suggestion, observation, context=None, metadata=metadata)
 
         # loop for llamatune-optimizer
-        suggestion = llamatune_optimizer.suggest()
+        suggestion, metadata = llamatune_optimizer.suggest()
         _x, _y = suggestion['x'].iloc[0], suggestion['y'].iloc[0]
         assert _x == pytest.approx(_y, rel=1e-3) or _x + _y == pytest.approx(3., rel=1e-3)  # optimizer explores 1-dimensional space
         observation = objective(suggestion)
-        llamatune_optimizer.register(suggestion, observation)
+        llamatune_optimizer.register(suggestion, observation, context=None, metadata=metadata)
 
     # Retrieve best observations
     best_observation = optimizer.get_best_observations()
     llamatune_best_observation = llamatune_optimizer.get_best_observations()
 
-    for (best_config, best_score, best_context) in (best_observation, llamatune_best_observation):
+    for (best_config, best_score, _, _) in (best_observation, llamatune_best_observation):
         assert isinstance(best_config, pd.DataFrame)
         assert isinstance(best_score, pd.DataFrame)
-        assert best_context is None
         assert set(best_config.columns) == {'x', 'y'}
         assert set(best_score.columns) == {'score'}
 
-    (best_config, best_score, _context) = best_observation
-    (llamatune_best_config, llamatune_best_score, _context) = llamatune_best_observation
+    (best_config, best_score, _, _) = best_observation
+    (llamatune_best_config, llamatune_best_score, _metadata) = llamatune_best_observation
```

Comment:
I am confused. shouldn't it be 
```suggestion
    (llamatune_best_config, llamatune_best_score, _context, _metadata) = llamatune_best_observation
```
?
this should fail in tests